From 14a11c18c6ffe1694d4ac0fe4f5dc25685fbb658 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 056/296] AlarmManagerService: null check the alarm operation

If the operation is null it will lead to a crash.
---
 .../java/com/android/server/AlarmManagerService.java   | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/services/core/java/com/android/server/AlarmManagerService.java b/services/core/java/com/android/server/AlarmManagerService.java
index 9d3f6299d3c..022c431413b 100644
--- a/services/core/java/com/android/server/AlarmManagerService.java
+++ b/services/core/java/com/android/server/AlarmManagerService.java
@@ -2727,14 +2727,18 @@ class AlarmManagerService extends SystemService {
                     // now trigger the alarms without the lock held
                     for (int i=0; i<triggerList.size(); i++) {
                         Alarm alarm = triggerList.get(i);
-                        try {
-                            alarm.operation.send();
-                        } catch (PendingIntent.CanceledException e) {
-                            if (alarm.repeatInterval > 0) {
-                                // This IntentSender is no longer valid, but this
-                                // is a repeating alarm, so toss the hoser.
-                                removeImpl(alarm.operation);
+                        if (alarm.operation != null) {
+                            try {
+                                alarm.operation.send();
+                            } catch (PendingIntent.CanceledException e) {
+                                if (alarm.repeatInterval > 0) {
+                                    // This IntentSender is no longer valid, but this
+                                    // is a repeating alarm, so toss the hoser.
+                                    removeImpl(alarm.operation);
+                                }
                             }
+                        } else {
+                            removeImpl(alarm.operation);
                         }
                     }
                     break;
-- 
2.11.0

